source/extensions/filters/network/connection_limit/connection_limit.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <chrono>
+#include <thread>
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/event/timer.h"
+#include "envoy/extensions/filters/network/connection_limit/v3/connection_limit.pb.h"
+#include "envoy/network/filter.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "common/common/thread_synchronizer.h"
+#include "common/runtime/runtime_protos.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ConnectionLimitFilter {
+
+/**
+ * All connection limit stats. @see stats_macros.h
+ */
+#define ALL_CONNECTION_LIMIT_STATS(COUNTER, GAUGE)                                                 \
+  COUNTER(connection_limited)                                                                      \
```

Comment:
Nit: the stat names would more consistent if it was `limited_connections`.

Reply:
Renamed to limited_connections.